tests/phpunit/Checker/Default_Check_Repository_Tests.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Tests for the Default_Check_Repository class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\Check_Repository;
+use WordPress\Plugin_Check\Checker\Default_Check_Repository;
+use WordPress\Plugin_Check\Test_Data\Invalid_Check;
+use WordPress\Plugin_Check\Test_Data\Runtime_Check;
+use WordPress\Plugin_Check\Test_Data\Static_Check;
+
+class Default_Check_Repository_Tests extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->repository = new Default_Check_Repository();
+	}
+
+	public function test_register_static_check() {
+		$check = new Static_Check();
+		$this->repository->register_check( 'static_check', $check );
+
+		$this->assertSame( array( $check ), $this->repository->get_checks() );
+	}
+
+	public function test_register_runtime_check() {
+		$check = new Runtime_Check();
+		$this->repository->register_check( 'runtime_check', $check );
+
+		$this->assertSame( array( $check ), $this->repository->get_checks() );
+	}
+
+	public function test_register_exception_thrown_for_invalid_check() {
+		$this->expectException( 'Exception' );
+		$this->expectExceptionMessage( 'Check must be an instance of Runtime_Check or Static_Check.' );
+
+		$this->repository->register_check( 'empty_check', new Invalid_Check() );
+	}
+
+	public function test_register_exception_thrown_for_existing_check_slug() {
+		$this->expectException( 'Exception' );
+		$this->expectExceptionMessage( 'Check slug "static_check" is already in use.' );
+
+		$this->repository->register_check( 'static_check', new Static_Check() );
+		$this->repository->register_check( 'static_check', new Static_Check() );
```

Comment:
Nit-pick: Maybe use a `Runtime_Check` in one of the two, to ensure the conflicts are also checked against the list of the respective other type? Or add a separate unit test for it, either works.

Reply:
I've included a new test for runtime checks

Reply:
@jjgrainger My point here was not to have another check for `Runtime_Check` checks specifically, but rather to ensure the slug check works also _between_ the two types of checks. E.g. if you first register a static check with a slug `my_slug`, and then register a runtime check with slug `my_slug`, it should still throw the exception, even though the checks are of a different type.

Reply:
@felixarntz Ah, I understand now. I've added a new test to capture this too.